nexus/src/nexus.rs

```diff
@@ -577,15 +579,19 @@ impl Nexus {
 
     pub async fn project_fetch(
         &self,
+        opctx: &OpContext,
         organization_name: &Name,
         project_name: &Name,
     ) -> LookupResult<db::model::Project> {
-        let organization_id = self
+        let authz_org = self
+            .db_datastore
+            .organization_lookup_path(organization_name)
+            .await?;
+        Ok(self
             .db_datastore
-            .organization_lookup_id(organization_name)
+            .project_fetch(opctx, &authz_org, project_name)
             .await?
-            .id();
-        self.db_datastore.project_fetch(&organization_id, project_name).await
+            .1)
```

Comment:
The `.1` here feels a little opaque. One alternative would be to use a struct instead of a tuple for the lookup result (so here you would do `.model` or something), but maybe that's too heavy.